### DISCUSSION TITLE
Either the typing for `RouteStatistics` is incorrect or there is a bug in the statistics logic?

### DISCUSSION THREAD

**Discussion by raman325** [zwave-js/zwave-js#6608] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.
- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

#### What is happening?

As reported in https://github.com/home-assistant/core/issues/108021 the HA integration expects `protocolDataRate` to be provided in the `lwr` and `nlwr` statistics objects based on the typing: https://github.com/zwave-js/node-zwave-js/blob/master/packages/zwave-js/src/lib/node/NodeStatistics.ts#L56

So to me it sounds like either the typing is wrong or there is some state that statistics get put in that is incorrect.

#### What did you expect to happen instead?

I think that either the typing needs to be updated to make `protocolDataRate` optional or the logic that is leaving the object in a bad state is updated.

#### Steps to reproduce the behavior:

Hard to say since it is happening on initialization. It's also not a widespread error as this is the first report we've seen.

#### Anything else we should know?

The user mentioned that it was not an issue in HA 2023.9.X which was released on September 6th. Looking at the `zwave-js-server` release logs, that suggests that this was introduced in zwave-js 12.x. I did try to run 2023.9.3 using the latest `node-zwave-js` just to make sure it wasn't some sort of backwards compatibility issue, but I was able to bring up my network fine

cc @maheus

### Software versions

Driver (node-zwave-js): 12.4.1 
Z-Wave JS UI: 9.6.2.6e369a1

Home Assistant Z-Wave Integration: Home Assistant 2023.11.3

Home Assistant Z-Wave JS Addon: ...

ioBroker.zwave2 Adapter: ...

If you are using something non-standard, tell us here: ...

### Device information

Unknown at this time

Manufacturer: ...
Model name: ...
Node ID: ...

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.
- [ ] The log includes a re-interview of the problematic device _(if applicable)_.
- [ ] The log includes the problematic interaction with the device _(if applicable)_.
- [ ] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

Waiting for the user to provide debug logs as they have stated they are worried about leaking personal information, but logging the issue anyway as a placeholder

[a comment by an automated account is not shown]

**Comment by AlCalzone** [comment]
According to that issue, the statistics only contained the repeaters. The only way I can imagine this to happen is if the response to GetPriorityRoute is one byte short (e.g. caused by a controller bug), and therefore the `routeSpeed` is `undefined` here:
https://github.com/zwave-js/node-zwave-js/blob/9549a040477364b54a2df617a42fd30639e67e14/packages/zwave-js/src/lib/serialapi/network-mgmt/GetPriorityRouteMessages.ts#L87-L92

Could that be? Does HA even use this method?

**Reply by raman325** [reply]
no we don't use this, at least not directly. This is happening on initial startup on the client side so this is coming from the initial state (we grab the statistics attribute directly from each node as part of the state). Just looking through the code, this is coming from `StatisticsHost` which the mixin for the Node class `NodeStatisticsHost` extends. I don't know if that's helpful or not, just not sure where to go from here

**Reply by kpine** [reply]
It looks like zwave-js will grab a "route cache" from the NVM? Could it have bad data?

https://github.com/zwave-js/node-zwave-js/blob/fb93b59c2dcccf687afd58222509191c90022b48/packages/nvmedit/src/convert.ts#L365-L380

**Reply by AlCalzone** [reply]
> initial startup on the client side 

Does that mean Z-Wave JS UI was already running? In this case getPriorityRoute may be getting called.

Will need the driver logs to confirm though

**Comment by maheus** [comment]
Hi @AlCalzone  my log is already in debug, but the file is very big. Can I give you the last 1000 line or first 1000 lines ?

**Reply by AlCalzone** [reply]
Can you zip it and send the entire thing? Shouldn't be too big after zipping.

**Reply by maheus** [reply]
no i think after gzip, it will ok

**Reply by maheus** [reply]
this log provide by zwave-js-ui: 8.15.0.f48c52b , i going to upgrade to latest version and i going to upload log after
with zwave-js-ui: 8.15.0.f48c52b statistic does not have `lwr` field in statistics 

```
  "statistics": {
    "commandsTX": 1,
    "commandsRX": 0,
    "commandsDroppedRX": 0,
    "commandsDroppedTX": 0,
    "timeoutResponse": 0,
    "rtt": 296.8
  },
```

after upgrade `lwr` appears and Homeassistant return an error

**Reply by maheus** [reply]
```2024-01-17T21:41:31.198Z DRIVER « [RES] [GetPriorityRoute]
                                    node ID:     49
                                    route kind:  LWR
                                    repeaters:   none
                                    route speed: unknown (undefined)
```

[zwavejs_2024-01-17.log.gz](https://github.com/zwave-js/node-zwave-js/files/13969329/zwavejs_2024-01-17.log.gz)

**Reply by maheus** [reply]
all nodes has the value `unknown (undefined)` for   the field `route speed`

**Reply by maheus** [reply]
@AlCalzone my logs helps you ?

**Reply by raman325** [reply]
please be patient, he is on paternity leave and is not paying attention to github like he normally would

**Reply by maheus** [reply]
yes no problem

**Comment by AlCalzone** [comment]
Thanks, that confirms my suspicion. Very old Z-Wave SDK on the stick, apparently does not include the byte for the route speed.

**Reply by AlCalzone** [reply, marked as answer]
Escalated to an issue: https://github.com/zwave-js/node-zwave-js/issues/6616

@raman325 seems like applications have to consider the `protocolDataRate` field to be optional.
